soc/nxp/imxrt/mpu_regions.c

```diff
@@ -27,6 +31,17 @@ static const struct arm_mpu_region mpu_regions[] = {
 	 */
 	MPU_REGION_ENTRY("SDRAM0", SDRAM_BASE_ADDR, REGION_IO_ATTR(REGION_512M)),
 #endif
+
+#ifdef CONFIG_SOC_MIMXRT1189_CM7
+#ifndef CONFIG_FLASH_MCUX_FLEXSPI_XIP
+	/*
+	 * Region 3 - mark external flash used flexspi1 as FLASH attribution
+	 * when boot isn't performed from external flash on cm7 core.
+	 * Otherwise cache maintenance will occur error in flash driver.
+	 */
+	MPU_REGION_ENTRY("EXT_FLASH0", EXT_FLASH0_BASE_ADDR, REGION_FLASH_ATTR(REGION_128M)),
```

Comment:
I think you can also declare the mpu region in DT, is there a reason it's hardcoded here

Reply:
Yes, today I review the case, I find the coot course why run failed on cm7 should be caused by the cm7 core speculative prefetches. So there is no need to add mpu configuration for flash memory.